### PR TITLE
gh-102890: Fix error handling in posix.stat implementation

### LINKED ISSUE #102890: Lack of error handling in `posix.stat` implementation leads to crash on interrupt
`_pystat_fromstructstat` function doesn't check that calls to `_pystat_l128_from_l64_l64` or `fill_time` didn't fail and/or exception wasn't set on current `tstate`, so subsequent calls to `fill_time` can lead to crash. I can't find less complicated reproducer yet, since it may require some exotic and perhaps nonexistent data to be passed to `posix.stat`.
Reproduced on current main (76350e85ebf96df588384f3d9bdc20d11045bef4), platform is linux (x86_64).

```python
import posix
import random
import threading
import _thread
import time

def crasher_thread():
    time.sleep(0.000001 * random.randint(1, 30))
    _thread.interrupt_main()

def try_crash():
    try:
        ct = threading.Thread(target=crasher_thread)
        ct.start()
        posix.stat(__file__)  # filename of any existing file
        ct.join()
    except KeyboardInterrupt:
        pass

for i in range(1000):
    try_crash()
```

Output:
```
Fatal Python error: _Py_CheckSlotResult: Slot * of type int succeeded with an exception set
Python runtime state: initialized
KeyboardInterrupt

Current thread 0x00007f1a12016740 (most recent call first):
  File "/home/.../repro.py", line 15 in try_crash
  File "/home/.../repro.py", line 21 in <module>
Aborted (core dumped)
```

I'll submit a PR with possible fix.

### Linked PRs
* gh-102891

### PR DESCRIPTION
**Pull request by chgnrdv**
Fixes #102890

* Issue: gh-102890

### CODE REVIEW

File: Lib/test/test_posix.py [hunk 1]
```diff
@@ -647,6 +647,33 @@ def test_stat(self):
                 'should be string, bytes, os.PathLike or integer, not',
                 posix.stat, list(os.fsencode(os_helper.TESTFN)))
 
+        # see gh-102890
+        code = f"""if 1:
```
Comment (sobolevn) [inline Lib/test/test_posix.py:652]: This should be a separate test case, please add a new `test_` method.

File: Modules/posixmodule.c [hunk 2]
```diff
@@ -2394,9 +2395,8 @@ fill_time(PyObject *module, PyObject *v, int s_index, int f_index, int ns_index,
         goto exit;
 
     float_s = PyFloat_FromDouble(sec + 1e-9*nsec);
-    if (!float_s) {
+    if (!float_s)
```
Comment (sobolevn) [inline Modules/posixmodule.c]: I think that style-only change here should be reverted. Diff is more important than braces position.

**Review by sobolevn — COMMENTED** [review]
Please, also add a NEWS entry.

**Comment by chgnrdv** [comment]
https://github.com/python/cpython/blob/d1a89ce5156cd4e1eff5823ec2200885c43e395e/Modules/posixmodule.c#L2479
Looks like this value returned by `PyLong_FromLongLong` should be NULL-checked too.

File: Modules/posixmodule.c (hunk 2 above)
Comment (chgnrdv) [reply Modules/posixmodule.c]: Done. I thought that I've added this one :-)

File: Lib/test/test_posix.py (hunk 1 above)
Comment (chgnrdv) [reply Lib/test/test_posix.py:652]: Done.

**Comment by chgnrdv** [comment]
Sorry for mess, I didn't notice earlier that this NULL check is enough. https://github.com/python/cpython/blob/420bbb783b43216cc897dc8914851899db37a31d/Modules/posixmodule.c#L2479

**Comment by chgnrdv** [comment]
The checks in `_pystat_fromstructstat` look exhaustive now.

**Comment by chgnrdv** [comment]
@sobolevn, could you review this PR again?

**Pull request closed without merging**